crates/humanode-peer/src/service/inherents.rs

```diff
@@ -104,3 +104,39 @@ impl<Client> Clone for Creator<Client> {
         }
     }
 }
+
+/// Sealing case related inherent data providers.
+pub mod sealing {
+    use std::cell::RefCell;
+
+    thread_local!(static TIMESTAMP: RefCell<u64> = RefCell::new(0));
```

Comment:
You can't use thread local values in the async context, because async fns can be scheduled to the async runtime as tasks, which means they'll execute on a thread pool, meaning each thread will have its own value - and the values will not be shared. If you need a local variable - just use a static variable with an access serialization method like mutex, or an atomic (since it is a primitive integer).